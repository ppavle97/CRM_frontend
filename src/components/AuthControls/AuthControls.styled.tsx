import styled from "@emotion/styled";
import { breakpointMax, calcRem, colors } from "../../theme";

export const StyledAuth = styled("div")({
  width: "100%",
  display: "flex",
  justifyContent: "center",
  "& button": {
    width: calcRem(200),
    height: calcRem(50),
    fontSize: calcRem(18),
    fontWeight: 600,
    margin: `0 ${calcRem(10)}`,
    border: "none",
    borderRadius: calcRem(10),
    color: colors.white,
    [breakpointMax("lg")]: {
      width: calcRem(180),
      height: calcRem(45),
    },
    [breakpointMax("sm")]: {
      width: calcRem(150),
      height: calcRem(45),
      fontSize: calcRem(16),
    },
    "&:hover": { cursor: "pointer" },
    "&:nth-child(1)": {
      backgroundColor: colors.teal,
    },
    "&:nth-child(2)": {
      backgroundColor: colors.purple,
    },
  },
});
