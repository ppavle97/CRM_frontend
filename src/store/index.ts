import { configureStore } from "@reduxjs/toolkit";
import modalReducer from "./features/modal/modalSlice";

const store = configureStore({
  reducer: {
    modal: modalReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
