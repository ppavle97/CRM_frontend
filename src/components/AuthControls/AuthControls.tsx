import { setModal } from "../../store/features/modal/modalSlice";
import { useAppDispatch } from "../../store/hooks";
import { StyledAuth } from "./AuthControls.styled";

export const AuthControls = () => {
  const dispatch = useAppDispatch();

  return (
    <StyledAuth>
      <button onClick={() => dispatch(setModal("login"))}>Login</button>
      <button onClick={() => dispatch(setModal("register"))}>Register</button>
    </StyledAuth>
  );
};
