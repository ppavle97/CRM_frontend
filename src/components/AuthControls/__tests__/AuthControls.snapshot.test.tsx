import { render } from "@testing-library/react";
import { AuthControls } from "../AuthControls";
import { useAppDispatch } from "../../../store/hooks";

jest.mock("../../../store/hooks", () => ({
  useAppDispatch: jest.fn(),
}));

describe("AuthControls component", () => {
  it("matches snapshot", () => {
    const mockDispatch = jest.fn();
    (useAppDispatch as jest.Mock).mockReturnValue(mockDispatch);

    const { asFragment } = render(<AuthControls />);

    expect(asFragment()).toMatchSnapshot();
  });
});
