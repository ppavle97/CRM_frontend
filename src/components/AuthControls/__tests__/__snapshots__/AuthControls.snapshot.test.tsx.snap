// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthControls component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-2oq9ft"
  >
    <button>
      Login
    </button>
    <button>
      Register
    </button>
  </div>
</DocumentFragment>
`;
